File: dcp-solutions/2024-10-14/solution.go
```markdown
func sortCharacters(s string) {
    counts := map[rune]int{}
    for _, c := range s {
        counts[c]++
    }

    first, second, third := []rune{}, []rune{}, []rune{}
    for k := range counts {
        for i := 0; i < counts[k]; i++ {
            if k == 'R' {
                first = append(first, k)
            } else if k == 'G' {
                second = append(second, k)
            } else {
                third = append(third, k)
            }
        }
    }

    var result []rune
    for len(first) > 0 || len(second) > 0 || len(third) > 0 {
        if len(first) == 0 {
            result = append(result, second[0])
            second = second[1:]
        } else if len(second) == 0 {
            result = append(result, first[0])
            first = first[1:]
        } else if len(third) == 0 {
            result = append(result, third[0])
            third = third[1:]
        } else {
            if counts['R'] >= counts['G'] && counts['R'] > counts['B'] {
                result = append(result, 'R')
                counts['R']--
            } else if counts['G'] > counts['R'] || counts['G'] == counts['R'] && counts['G'] > counts['B'] {
                result = append(result, 'G')
                counts['G']--
            } else {
                result = append(result, 'B')
                counts['B']--
            }
        }
    }

    fmt.Println(string(result))
}
```

File: dcp-solutions/2024-10-14/solution.py
```python
def sort_characters(arr):
    """
    Segregate the values of the array so that all the Rs come first, 
    the Gs come second, and the Bs come last.
    
    Args:
        arr (list): A list of strictly the characters R, G, and B
    
    Returns:
        None
    """
    
    # Count the occurrences of each character
    r_count = 0
    g_count = 0
    b_count = 0
    for char in arr:
        if char == 'R':
            r_count += 1
        elif char == 'G':
            g_count += 1
        elif char == 'B':
            b_count += 1
    
    # Initialize pointers to track the current positions of Rs, Gs and Bs
    i = j = k = 0
    
    while i < len(arr) or j < len(arr) or k < len(arr):
        if i < len(arr) and arr[i] == 'R':
            arr[j], arr[k] = arr[k], arr[j]
            r_count -= 1
            j += 1
            k += 1
        elif j < len(arr) and arr[j] == 'G':
            arr[i], arr[j] = arr[j], arr[i]
            g_count -= 1
            i += 1
            j += 1
        else:
            if r_count > 0 or g_count > 0:
                arr[k], arr[j] = arr[j], arr[k]
                b_count -= 1
                k += 1
            else:
                j += 1

# Test the function with [G, B, R, R, B, R, G]
arr = ['G', 'B', 'R', 'R', 'B', 'R', 'G']
sort_characters(arr)
print(arr)